testing/file_test/test_file.cpp

```diff
@@ -93,9 +96,53 @@ struct SplitState {
   int file_index = 0;
 };
 
+static auto ExtractFilePathFromUri(llvm::StringRef uri)
+    -> std::optional<llvm::StringRef> {
+  static constexpr llvm::StringRef FilePrefix = "file:/";
+  if (!uri.starts_with(FilePrefix)) {
+    return std::nullopt;
+  }
+  return uri.drop_front(FilePrefix.size());
+}
+
+static auto AutoFillDidOpenParams(llvm::json::Object* params,
+                                  llvm::ArrayRef<TestFile::Split> splits)
+    -> ErrorOr<Success> {
+  auto* text_document = params->getObject("textDocument");
+  if (text_document == nullptr) {
+    return Success{};
+  }
+
+  auto attr_it = text_document->find("text");
+  if (attr_it == text_document->end() || attr_it->second != "FROM_FILE_SPLIT") {
+    return Success{};
+  }
+
+  auto uri = text_document->getString("uri");
+  if (!uri) {
+    return ErrorBuilder() << "missing uri in params.textDocument";
```

Comment:
```suggestion
    return Error("missing uri in params.textDocument");
```

Minor -- no reason to stream with a simple string